### PR TITLE
snap_restore: Abort last recover region

### LINKED ISSUE #15684: snap_restore: there should only be one `recover_region` RPC.
## Development Task

cc https://github.com/pingcap/tidb-operator/issues/5297

Also note this solution assumes a individual client is always sane(i.e. won't assign two different stores as leader for one region).

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close #15684
What's Changed:

```commit-message
This PR will make `recover_region` return `ABORTED` once there are new `recover_region` RPCs in.
```

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Fixed a bug that may cause snapshot restore stuck when BR crashes.
```

### CODE REVIEW

File: components/snap_recovery/src/services.rs [hunk 1]
```diff
@@ -215,18 +277,23 @@ impl<ER: RaftEngine> RecoverData for RecoveryService<ER> {
             }
         });
 
+        // Hacking: Some times, the client may omit the RPC call to `recover_region` if
```
Comment (BornChanger) [inline components/snap_recovery/src/services.rs:280]:
```suggestion
        // Hacking: Sometimes, the client may omit the RPC call to `recover_region` if
```

File: components/snap_recovery/src/services.rs [hunk 2]
```diff
@@ -301,10 +370,21 @@ impl<ER: RaftEngine> RecoverData for RecoveryService<ER> {
...
             };
 
-            let _ = sink.success(resp).await;
+            resp
         };
 
-        self.threads.spawn_ok(task);
+        self.abort_last_recover_region(format!("recover_region by {}", ctx.peer()));
+        let (state, task) = RecoverRegionState::wrap_task(task);
+        *self.last_recovery_region_rpc.lock().unwrap() = Some(state);
```
Comment (3pointer) [inline components/snap_recovery/src/services.rs:378]: maybe move this line into `self.abort_last_recover_region`?
it seems still race here in some corner cases.

**Review by 3pointer — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Comment by BornChanger** [comment]
@YuJuncen please also cherry-pick to 6.5 and 7.1.

**Pull request merged**